Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1393,12 +1397,43 @@ void LocalFrame::networkUsageDidExceedThreshold()
 
     FRAME_RELEASE_LOG_ERROR(ResourceLoading, "networkUsageDidExceedThreshold: Unloading frame due to exceeding threshold.");
 
-    showResourceMonitoringError("This frame was unloaded."_s);
+    showResourceMonitoringError(generateResourceMonitorErrorHTML());
 
     if (RefPtr document = this->document())
         document->addConsoleMessage(MessageSource::ContentBlocker, MessageLevel::Error, "Frame was unloaded because its network usage exceeded the limit."_s);
 }
 
+String LocalFrame::generateResourceMonitorErrorHTML() const
+{
+#if PLATFORM(COCOA) && HAVE(LOCAL_FRAME_ADDITIONS)
+    return generateResourceMonitorErrorHTMLForCocoa();
+#else
+    return String::fromUTF8(R"(<meta charset="utf-8" />)"
+        "<style>"
+        "body {"
+            "margin:0;padding:0;"
+            "background-color:rgb(242,242,247);"
+            "color:rgb(139,139,143);"
+            "font-family:system-ui,sans-serif; font-size:0.8rem; text-align:center;"
+        "}"
+        "a {"
+            "width:100%; height:100%;"
+            "display:flex; align-items:center; justify-items:center;"
+            "cursor:help;"
+        "}"
+        "span { margin:0 auto; }"
+        "@media (width<60px) { .L,.M { display:none; } }"
+        "@media (60px<=width<160px) { .L,.S { display:none; } }"
+        "@media (width>=160px) { .S,.M { display:none; } }"
+        "</style>"
+        R"(<a title="This frame is hidden for using too many system resources.">)"
+        "<span class=L>⛔︎ Frame hidden for using significant resources</span>"
+        "<span class=M>⛔︎<br />Frame hidden</span>"
+        "<span class=S>⛔︎</span>"
+        "</a>");
```

Comment:
I think just returning "This frame was unloaded for using too many system resources" is probably fine here for the `#else` case. Other platforms can figure out how to customize this when they get to that point.